src/merkle_tree.cairo

```diff
@@ -0,0 +1,88 @@
+use super::utils::double_sha256;
+
+pub fn merkle_root(ref txids: Array<u256>) -> u256 {
+    let len = txids.len();
+    if len % 2 == 1 {
+        txids.append(*txids.at(len - 1));
+    } else {
+        // CVE-2012-2459 bug fix
+        assert!(
+            txids.at(len - 1) != txids.at(len - 2), "unexpected node duplication in merkle tree"
+        );
+    }
+
+    let mut next_txids = ArrayTrait::new();
+    let mut i = 0;
+    while i < len {
+        next_txids.append(double_sha256(*txids.at(i), *txids.at(i + 1)));
+        i += 2;
+    };
+
+    merkle_root(ref next_txids)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{merkle_root};
+    use starknet::core::types::FieldElement;
+
+    #[test]
+    fn test_merkle_root() {
+        let txids = vec![
+            FieldElement::from_hex_be(
+                "50ba87bdd484f07c8c55f76a22982f987c0465fdc345381b4634a70dc0ea0b38"
```

Comment:
Why don't you just declare it as `0x50ba87bdd484f07c8c55f76a22982f987c0465fdc345381b4634a70dc0ea0b38_u256`?

Reply:
declare it as constant ?

Reply:
You should be able to declare u256 Array like:
```
 array![0x50ba87bdd484f07c8c55f76a22982f987c0465fdc345381b4634a70dc0ea0b38_u256, 0x96b8787b1e3abed802cff132c891c2e511edd200b08baa9eb7d8942d7c5423c6_u256,...]
```
using integer literals.